### ISSUE TITLE
Typos in 01 intro to pytorch

### ISSUE THREAD

**Issue by marmatti** [Edinburgh-Chemistry-Teaching/ATCP_23_24#2]
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/3250fb33-38fb-434f-8e11-c01818c1c2b3)

Deep neural network built on **an(?)** autograd system

-----------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/c0f622ee-02f2-4d50-8b69-3d43df9877c4)
 
The following cell **installs(?)** necessary packages...

--------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/fe20fda5-956d-4529-83fd-08a166bef502)

...and **it(?)** is just a generic n-dimensional array to be used...

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/63d36128-fe1c-4e40-8b78-0025df6703e5)

#We can also specify **a(?)** data type

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/a37d79ed-0edc-4362-bdb6-89c6257d244f)

Casting tensor x as **a(?)** CUDA datatype if CUDA **is(?)** available 

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/e40d0819-7a7e-4cba-85bb-0bf4f57042b5)

Attention: x.to(device) will not cast x as **a(?)** cuda datatype - we need x = x.to(device)

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/bf8d706e-c2bf-4ba7-9706-905eb7136a42)

**I do not think this flows super clearly. I would rewrite:**
Note: **As mentioned above, whenever we work with tensors,** x.to(device) does not move x to cuda and we need to write x = x.to(device) instead.

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/e8328f63-51fc-49aa-8492-6fef64a63dfa)

DoubleTensor is **a(?)** 64-bit floating point and FloatTensor is **a(?)** 32-bit floating point tensor. So a FloatTensor **uses half of the memory as a DoubleTensor with the same size (?)** ... So PyTorch leaves it **up(?)** to **the(?)** user to choose...

---------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/da59ccaf-51c2-44fe-9190-9d0659c63b03)

Set **a** default tensor type for your notebook

----------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/3dee2895-8ed6-4a8d-85b6-28480a887353)

The MNIST **hand**-written digits dataset consists of 60,000 training...

--------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/2fd9ef33-5b4d-4135-8900-7deee0fca657)

...Since we have been using **scikit-learn,** we will use **the(?)** **dataset** in the example here:
**Dataset and data set equally valid for spelling, just a suggestion to uniform.**

--------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/d8720986-84c6-4987-bff6-59e0366b9b98)

Splitting the data into train and test **sets**

--------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/32acfd75-ef5d-4f69-81bf-89e2a713f5ef)

... This is just an example **full stop "." missing**

---------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/72f9deda-1be2-4ebe-bf01-9822d1bb17c7)

If np.argmax(out,axis=1)-y is non-zero, **(I would move this here -->) i.e. every time there is a mismatch between prediction and label, e.g.(?)** label y is 9 and prediction is 5 **comma(?)** then diff is incremented by 1. 
    
----------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/85d058a0-99c0-4cb4-9c03-2d93887b5a3f)

we can't call numpy() on Tensors that **require** grad.

-----------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/7c22e37b-4ce0-4c54-aedf-5133aae7f0f0)

The optimiser module gives access to a large number of standard optimisers that try **to** help minimise the loss.

-----------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/d975a7db-5c07-4152-bd25-de4bf0b9be49)

Typically for a classification problem one would use a cross **entropy** loss **:(Colon?)** the torch documentation has some more details on this.

---------------------------------
![image](https://github.com/Edinburgh-Chemistry-Teaching/ATCP_23_24/assets/61904349/2908985a-abd2-4478-902d-1ffcbd2b2e18)

This is a parameter you set in your optimiser **and** you can play around with different values for it.

----------------------------------

**Comment by marmatti** [comment]
Sorry I checked data set and dataset are equally valid as spelling

**Comment by ppxasjsm** [comment]
@ryankzhu can you fix these please thanks!
